pkg/backup/rbac.go

```diff
@@ -0,0 +1,152 @@
+	// List returns a slice of ClusterRoleBindings which can represent either v1 or v1beta1 ClusterRoleBindings.
+	List() ([]ClusterRoleBinding, error)
+}
+
+type v1ClusterRoleBindingLister struct {
+	client rbacclient.ClusterRoleBindingInterface
+}
+
+func (v1 v1ClusterRoleBindingLister) List() ([]ClusterRoleBinding, error) {
+	crbList, err := v1.client.List(metav1.ListOptions{})
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	var crbs []ClusterRoleBinding
+	for _, crb := range crbList.Items {
+		crbs = append(crbs, v1crb{crb: crb})
+	}
+
+	return crbs, nil
+}
+
+type v1beta1ClusterRoleBindingLister struct {
+	client rbacclientbeta.ClusterRoleBindingInterface
+}
+
+func (v1beta1 v1beta1ClusterRoleBindingLister) List() ([]ClusterRoleBinding, error) {
+
+	crbList, err := v1beta1.client.List(metav1.ListOptions{})
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	var crbs []ClusterRoleBinding
+	for _, crb := range crbList.Items {
+		crbs = append(crbs, v1beta1crb{crb: crb})
+	}
+
+	return crbs, nil
+}
+
+// NewClusterRoleBindingLister wraps the v1beta1 and v1 RBAC client's List with a common interface.
+func NewClusterRoleBindingLister(log logrus.FieldLogger, clientset kubernetes.Interface) (ClusterRoleBindingLister, error) {
+	discoveryHelper, err := arkdiscovery.NewHelper(clientset.Discovery(), log)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	var supportedAPI metav1.GroupVersionForDiscovery
+	for _, ag := range discoveryHelper.APIGroups() {
+		if ag.Name == rbac.GroupName {
+			supportedAPI = ag.PreferredVersion
+			break
+		}
+	}
+
+	switch supportedAPI.Version {
+	case rbac.SchemeGroupVersion.Version:
+		return v1ClusterRoleBindingLister{client: clientset.RbacV1().ClusterRoleBindings()}, nil
+	case rbacbeta.SchemeGroupVersion.Version:
+		return v1beta1ClusterRoleBindingLister{client: clientset.RbacV1beta1().ClusterRoleBindings()}, nil
+	// Return an empty client if we don't have a known version.
+	default:
+		return nil, nil
+	}
+}
+
+// ClusterRoleBinding abstracts access to ClusterRoleBindings whether they're v1 or v1beta1.
+type ClusterRoleBinding interface {
+	// Name returns the name of a ClusterRoleBinding.
+	Name() string
+	// ServiceAccountSubjects returns the names of subjects that are service accounts in the given namespace.
+	ServiceAccountSubjects(namespace string) []string
+	// RoleRefName returns the name of a ClusterRoleBinding's RoleRef.
+	RoleRefName() string
+}
+
+// v1 implementation
+type v1crb struct {
+	crb rbac.ClusterRoleBinding
+}
+
+func (c v1crb) Name() string {
+	return c.crb.Name
+}
+
+func (c v1crb) RoleRefName() string {
+	return c.crb.RoleRef.Name
+}
+
+func (c v1crb) ServiceAccountSubjects(namespace string) []string {
+	var saSubjects []string
+	for _, s := range c.crb.Subjects {
+		if s.Kind == rbac.ServiceAccountKind && s.Namespace == namespace {
+			saSubjects = append(saSubjects, s.Name)
+		}
+	}
+	return saSubjects
+}
+
+// v1beta1 implementation
```

Comment:
I'd remove this, put a godoc comment on the struct if you want but not necessary